pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -757,21 +757,18 @@ func applyOutlierDetection(c *cluster.Cluster, outlier *networking.OutlierDetect
 }
 
 func applyLoadBalancer(c *cluster.Cluster, lb *networking.LoadBalancerSettings, port *model.Port, proxy *model.Proxy, meshConfig *meshconfig.MeshConfig) {
-	lbSetting := loadbalancer.GetLocalityLbSetting(meshConfig.GetLocalityLbSetting(), lb.GetLocalityLbSetting())
-	if c.OutlierDetection != nil {
+	localityLbSetting := loadbalancer.GetLocalityLbSetting(meshConfig.GetLocalityLbSetting(), lb.GetLocalityLbSetting())
+	if localityLbSetting != nil && (localityLbSetting.Distribute != nil || localityLbSetting.Failover != nil) {
```

Comment:
I come to think probably we donot need this check at all. Because in istio all the load balancing is based on locality weight even no `localityLbSetting`.

Typically we do change lb weight in pilot/pkg/xds/ep_filters.go, but trying to get to find if there are some integration tests for lb weight

Reply:
We have this kind of test tests/integration/pilot/locality_test.go. but it is with outlier detection

Reply:
Users can disable locality loadbalancer altogether https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go#L34 - I think we need to have this check

Reply:
This is different dimension stuff, it is whether to enable the settings in LocalityLoadBalancerSetting.  I am saying that envoy locality load balancing weight may  not be respected now when outlier detection is unset.

Reply:
I am not fully following this - Are you saying we should remove this 

`if localityLbSetting != nil && (localityLbSetting.Distribute != nil || localityLbSetting.Failover != nil) {`

and enable locality LB unconditionally?

Reply:
yes, i think we should do this

Reply:
Ok. Changed PTAL

Reply:
This will break users not using locality LB. I forget the details but i am certain Karl reported this ~1 year ago

Reply:
I think it was something like any pods that don't report locality will never get sent any traffic

Reply:
Ok.  Good point. I have added the check back.